File: Section Reports/Data/IListBinding/VB.NET/IListBinding/IlistReportSample.vb
Imports GrapeCity.ActiveReports.Document
Imports GrapeCity.ActiveReports.Document.Section
Imports GrapeCity.ActiveReports.SectionReportModel
Imports GrapeCity.ActiveReports.Controls
Imports GrapeCity.ActiveReports
Imports System
Public Class IlistReportSample
	Inherits GrapeCity.ActiveReports.SectionReport
	Private _productCollection As IListBinding.DataLayer.ProductCollection
	Private _greenBarCounter As Long = 0
	Sub New(productCollection As IListBinding.DataLayer.ProductCollection)
		_productCollection = productCollection
		InitializeComponent()
	End Sub
	
	Private Sub Detail_Format(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Detail.Format
		' Green bar effect colors the background of every other row.
		_greenBarCounter = _greenBarCounter + 1
		If _greenBarCounter Mod 2 = 0 Then
			Me.Detail.BackColor = System.Drawing.Color.Transparent
		Else
			Me.Detail.BackColor = System.Drawing.Color.LightGray
		End If
		' Reorder Row Coloring - Highlights the products that are to be
		'  re-ordered for a better understanding.
		'  Values for calculating the highlight.
		Dim _reorderLevel As Integer = Int32.Parse(Me.txtReorderLevel.Text)
		Dim _unitsInStock As Integer = Int32.Parse(Me.txtUnitsInStock.Text)
		If _unitsInStock < _reorderLevel Then
			' Highlight since the re-stocking level has been met.
			Me.Detail.BackColor = System.Drawing.Color.LightCoral
			
			Me.txtComments.Text = "Check " + Me.txtProductName.Text
			
			'
		Else
			' Reset the comments text
			Me.txtComments.Text = ""
		End If
	End Sub
	Private Sub IlistReportSample_ReportStart(sender As Object, e As EventArgs) Handles MyBase.ReportStart
		' Assign the productCollection DataLayer Object to the report's datasource
		Me.DataSource = _productCollection
	End Sub
End Class

File: Section Reports/Data/IListBinding/C#/IListBinding/IlistReportSample.cs
using System;
namespace GrapeCity.ActiveReports.Samples.IListBinding
{
	public partial class IlistReportSample : GrapeCity.ActiveReports.SectionReport
	{
		private long _greenBarCounter;
		private IListBinding.DataLayer.ProductCollection _productCollection;
		public IlistReportSample(IListBinding.DataLayer.ProductCollection _productCollection)
		{
		   
			this._productCollection = _productCollection;
			InitializeComponent();
			
			Document.Name = "IList Sample ActiveReport";
			
			
		}
		void IlistReportSample_ReportStart(object sender, EventArgs e)
		{
			// Assign the productCollection DataLayer object to the report's datasource
			this.DataSource = _productCollection;
		}
		private void Detail_Format(object sender, EventArgs eArgs)
		{
			// Green bar effect colors the background of every other row.
			_greenBarCounter++;
			if ((_greenBarCounter % 2) == 0)
			{
				// Even number rows
				Detail.BackColor = System.Drawing.Color.Transparent;
			}
			else
			{
				// Odd number rows
				Detail.BackColor = System.Drawing.Color.LightGray;
			}
			// Reorder Row Coloring - Highlights the products that are to be
			// re-ordered for a better understanding.
			// Values for calculating the highlight.
			int reorderLevel = Int32.Parse(txtReorderLevel.Text);
			int unitsInStock = Int32.Parse(txtUnitsInStock.Text);
			if (unitsInStock < reorderLevel)
			{
				// Highlight since the re-stocking level has been met.
				Detail.BackColor = System.Drawing.Color.LightCoral;
				
				txtComments.Text = "Check " + txtProductName.Text;
				
				
			}
			else
			{
				// Reset the comments text
				txtComments.Text = "";
			}
		}
	}
}
